sdk/cosmosdb/cosmos/samples/v4/javascript/ClientSideEncryption.js

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @summary Client-Side Encryption
+ */
+
+require("dotenv").config();
+
+const { ClientSecretCredential } = require("@azure/identity");
+
+const {
+  CosmosClient,
+  AzureKeyVaultEncryptionKeyResolver,
+  EncryptionKeyResolverName,
+  EncryptionTimeToLive,
+  EncryptionAlgorithm,
+  KeyEncryptionAlgorithm,
+  EncryptionKeyWrapMetadata,
+  ClientEncryptionIncludedPath,
+  EncryptionType,
+  ClientEncryptionPolicy,
+  EncryptionQueryBuilder,
+} = require("@azure/cosmos");
+const { finish, handleError, logStep } = require("./Shared/handleError");
+
+const key = process.env.COSMOS_KEY || "<cosmos key>";
+const endpoint = process.env.COSMOS_ENDPOINT || "<cosmos endpoint>";
+const databaseId = process.env.COSMOS_DATABASE || "<cosmos database>";
+const containerId = process.env.COSMOS_CONTAINER || "<cosmos container>";
+var client;
+
+async function run() {
+  logStep("Create encryption enabled cosmos client");
+  const credentials = new ClientSecretCredential("<tenant-id>", "<client-id>", "<app-secret>");
+
+  // create AzureKeyVaultEncryptionKeyResolver with credentials to get access to azure key vault
+  const keyResolver = new AzureKeyVaultEncryptionKeyResolver(credentials);
```

Comment:
Instead of inventing a new concept can the customer pass in their own [CryptographyClient](https://learn.microsoft.com/en-us/javascript/api/@azure/keyvault-keys/cryptographyclient?view=azure-node-latest) or a KeyClient + keyName which can be used by cosmos to [getCryptographyClient](https://learn.microsoft.com/en-us/javascript/api/@azure/keyvault-keys/keyclient?view=azure-node-latest#@azure-keyvault-keys-keyclient-getcryptographyclient)

This allows customers to use concepts that they are likely already familiar with, or at least are pre-existing

Reply:
`AzureKeyVaultEncryptionKeyResolver` wrapper allows resolving Keys from multiple instances of Azure KeyVault. Passing a KeyClient itself would restrict them to store keys in one key vault only.